src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/AdjudicationsResourceTest.java

```diff
@@ -605,4 +618,190 @@ public void deleteHearing() {
         }
     }
 
+    @Nested
+    public class CreateHearingResult {
+
+        final List<String> valid = List.of("ROLE_MAINTAIN_ADJUDICATIONS");
+        final List<String> invalid = List.of("ROLE_SYSTEM_USER");
+
+        final Map invalidRequest = Map.of("pleaFindingCode", PleaFindingCode.GUILTY, "findingCode", FindingCode.NOT_PROCEED, "adjudicator", "TWRIGHT");
+        final Map validRequest = Map.of("pleaFindingCode", PleaFindingCode.GUILTY, "findingCode", FindingCode.NOT_PROCEED, "adjudicator", "ITAG_USER");
+
+        @Test
+        public void createHearingResultReturns403ForInvalidRoles () {
+            createHearingResult(invalid, validRequest, -9L, -1L)
+                .expectStatus().isForbidden();
+        }
+
+        @Test
+        public void createHearingResultReturns404DueToNoAdjudication() {
+            createHearingResult(valid, validRequest, 99L, -1L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void createHearingResultReturns404DueToNoHearing() {
+            createHearingResult(valid, validRequest, -9L, 2L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void createHearingResultReturns404DueToNoAdjudicatorOnFile() {
+            createHearingResult(valid, invalidRequest, -9L, -4L)
+                .expectStatus().isNotFound();
+        }
+
+        @Test
+        public void createHearingResultReturns400DueToHearingNotBeingAssociatedWithAdjudication() {
+            createHearingResult(valid, validRequest, -5L, -4L)
+                .expectStatus().isBadRequest();
+        }
+
+        @Test
+        public void createHearingResultReturns400DueToHearingResultPresent() {
+            createHearingResult(valid, validRequest, -7L, -1L)
+                .expectStatus().isBadRequest();
+        }
+
+        @Test
+        public void createHearingResultReturnsSuccess() {
+            createHearingResult(valid, validRequest, -3001L, -3004L)
```

Comment:
maybe also call a GET to prove the result is really there rather than relying on response

Reply:
We don't have getHearingResult method, so cannot use it. Saving to DB operation is checked by service test using mock (so covered by test). Should I insert direct DB read here?